pkg/cmd/addon/util.go

```diff
@@ -40,3 +49,33 @@ func getAddonVersion(addon *extensionsv1alpha1.Addon) string {
 	}
 	return ""
 }
+
+func CheckBeforeDisableAddon(f cmdutil.Factory, addons []string) error {
+	labelSelecotor := util.BuildClusterLabel("", addons)
+	r := f.NewBuilder().
+		Unstructured().
+		AllNamespaces(true).
+		LabelSelector(labelSelecotor).
+		ResourceTypeOrNameArgs(true, []string{util.GVRToString(types.ClusterGVR())}...).
+		ContinueOnError().
+		Latest().
+		Flatten().
+		Do()
```

Comment:
I prefer to build a dynamic client by cmdutil.Factory and List the clusters, feels more intuitive to me personally.